instrumentation/spring/spring-boot-autoconfigure/build.gradle.kts

```
@@ -22,7 +21,7 @@ sourceSets {
 }
 
 dependencies {
-  implementation("org.springframework.boot:spring-boot-autoconfigure:$springBootVersion")
+  compileOnly("org.springframework.boot:spring-boot-autoconfigure:$springBootVersion")
```

would it work to change the base version to the real base version (2.6.0?), and change this to

```suggestion
  // needed to compile against optional AutoConfiguration annotation
  compileOnly("org.springframework.boot:spring-boot-autoconfigure:2.7.0")
```

and have the primary test be against the base version

and if you want to test the AutoConfiguration annotation code, then add a test suite to test against 2.7.0

---

- the smoke test is run against 2.6
- 2.6.0 doesn't work, it contains some issues

> and if you want to test the AutoConfiguration annotation code, then add a test suite to test against 2.7.0

it's tested with latest version

---

sorry, I didn't follow how it works and doesn't work at the same time? thanks

> * the smoke test is run against 2.6
> * 2.6.0 doesn't work, it contains some issues

---

seems it needs 2.7 to compile, but not at runtime

---

can we have compileOnly on 2.7, and testImplementation (or testLibrary) on 2.6?

---

tried that - results is runtime errors (` java.lang.NoSuchMethodError: 'reactor.core.publisher.Mono reactor.core.publisher.Mono.subscriberContext(reactor.util.context.Context)'`) which often happen if you have different spring versions

---

thanks for trying ❤️